soc/nordic/nrf54h/pm_s2ram.c

```diff
@@ -167,12 +167,12 @@ int soc_s2ram_suspend(pm_s2ram_system_off_fn_t system_off)
 	nvic_suspend(&backup_data.nvic_context);
 	mpu_suspend(&backup_data.mpu_context);
 	ret = arch_pm_s2ram_suspend(system_off);
+	/* Cache is powered down so power up is needed even if s2ram failed. */
+	nrf_power_up_cache();
 	if (ret < 0) {
 		return ret;
```

Comment:
This is probably not the problem, but just to be sure:
The only thing which won't be recovered here is MPU->RNR as it is overwritten [here](https://github.com/nrfconnect/sdk-zephyr/pull/2776/files#diff-2a4a5c7e5da1b79f4f9983df95da0ff9378c9d855390d4f0f90063d5b9fa8962R83). I don't think this register is relevant (it selects the currently accessed region) - if not then maybe we could remove it from [backup list](https://github.com/nrfconnect/sdk-zephyr/pull/2776/files#diff-2a4a5c7e5da1b79f4f9983df95da0ff9378c9d855390d4f0f90063d5b9fa8962R36)?